packages/dds/tree/src/shared-tree/sharedTree.ts

```diff
@@ -95,16 +94,49 @@ export interface ISharedTreeCheckout extends AnchorLocator {
 	/**
 	 * Current contents.
 	 * Updated by edits (local and remote).
-	 * Use `runTransaction` to create a local edit.
+	 * Use `runTransaction` to create a local edit. TODO
 	 */
 	readonly forest: IForestSubscription;
 
+	/**
+	 * Used to edit the state of the tree. Edits will be immediately applied locally to the tree.
+	 * If there is no transaction currently ongoing, then the edits will be submitted to Fluid immediately as well.
+	 */
+	readonly editor: IDefaultEditBuilder;
+
+	/**
+	 * An collection of functions for managing transactions. Transactions allow edits to be batched into atomic units.
+	 * Edits made during a transaction will update the local state of the tree immediately, but will be squashed into a single edit
+	 * when the transaction is committed. If the transaction is aborted, the local state will be reset to what it was before
+	 * the transaction began. Transactions may nest, meaning that a transaction may be started while a transaction is already
+	 * ongoing.
+	 *
+	 * TODO: Provide an option to avoid updating local state during a transaction.
```

Comment:
What would that option be used for?

Reply:
@taylorsw04 convinced me that we want this eventually. I think it's nice if there will be an intermediate state during your transaction which you would rather not project to your app UI. It's not a required API; you could also put conditionals in all your app's event listeners that listen to tree updates and lock with a global or something. But it seems like it'd be a nice feature to build into the transaction API itself, eventually.